src/Metadata/Resource/Factory/XmlResourceMetadataFactory.php

```diff
@@ -53,170 +53,63 @@ public function create(string $resourceClass) : ResourceMetadata
-
-        foreach (['shortName', 'description', 'itemOperations', 'collectionOperations', 'iri', 'attributes'] as $property) {
-            if (!isset($metadata[$property])) {
-                continue;
-            }
-
-            $resourceMetadata = $this->createWith($resourceMetadata, $property, $metadata[$property]);
-        }
-
-        return $resourceMetadata;
+        return null === $parentResourceMetadata ? new ResourceMetadata(...$metadata) : $this->update($parentResourceMetadata, $metadata);
     }
 
     /**
-     * Creates a DOMDocument based on `resource` tags of a file-loaded xml document.
+     * Extracts metadata from the XML tree.
      *
-     * @param string $path the xml file path
+     * @param string $resourceClass
      *
-     * @return \DOMDocument
+     * @return array
      */
-    private function getResourcesDom(string $path) : \DOMDocument
+    private function getMetadata(string $resourceClass) : array
     {
-        $doc = new \DOMDocument('1.0', 'utf-8');
-        $root = $doc->createElement('resources');
-        $doc->appendChild($root);
-
-        $this->xmlParser->loadXML(file_get_contents($path));
-
-        $xpath = new \DOMXpath($this->xmlParser);
-        $resources = $xpath->query('//resource');
-
-        foreach ($resources as $resource) {
-            $root->appendChild($doc->importNode($resource, true));
-        }
-
-        return $doc;
-    }
-
-    /**
-     * Get operations from xml.
-     *
-     * @param \DOMNodeList $query
-     *
-     * @return array|null
-     */
-    private function getOperations(\DOMNodeList $query)
-    {
-        $operations = [];
-        foreach ($query as $operation) {
-            $key = $operation->getAttribute('key');
-            $operations[$key] = [
-          'method' => $operation->getAttribute('method'),
-        ];
+        foreach ($this->paths as $path) {
+            try {
+                $domDocument = XmlUtils::loadFile($path, self::RESOURCE_SCHEMA);
+            } catch (\InvalidArgumentException $e) {
+                throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
+            }
 
-            $path = $operation->getAttribute('path');
+            $xml = simplexml_import_dom($domDocument);
+            foreach ($xml->resource as $resource) {
+                if ($resourceClass !== (string) $resource['class']) {
+                    continue;
+                }
 
-            if ($path) {
-                $operations[$key]['path'] = $path;
+                return [
+                    (string) $resource['shortName'] ?? null,
+                    (string) $resource['description'] ?? null,
+                    (string) $resource['iri'] ?? null,
+                    $this->getAttributes($resource, 'itemOperation') ?: null,
+                    $this->getAttributes($resource, 'collectionOperation') ?: null,
+                    $this->getAttributes($resource, 'attribute') ?: null,
```

Comment:
It just adds an element to the stack. Indeed, makes code slightly more complex but YAML, Annotations and PropertyMetadata are using the `attributes` keyword. As you want, I find both formats alright.